blas/unit_test/Test_Blas2_syr.hpp

```diff
@@ -0,0 +1,1725 @@
+
+  // ********************************************************************
+  // Call ger()
+  // ********************************************************************
+  std::cout << "In Test_Blas2_syr, '" << situation << "', alpha = " << alpha
+            << std::endl;
+  KOKKOS_IMPL_DO_NOT_USE_PRINTF(
+      "In Test_Blas2_syr.hpp, right before calling KokkosBlas::ger(): "
+      "ViewTypeA = %s, _kkGerShouldThrowException=%d\n",
+      typeid(_ViewTypeA).name(), _kkGerShouldThrowException);
+  std::string mode = _useHermitianOption ? "H" : "T";
+  bool gotStdException(false);
+  bool gotUnknownException(false);
+  try {
+    KokkosBlas::ger(mode.c_str(), alpha, x, x, A_ger.d_view);
+  } catch (const std::exception& e) {
+    std::cout << "In Test_Blas2_syr, '" << situation
+              << "', ger() call: caught exception, e.what() = " << e.what()
+              << std::endl;
+    gotStdException = true;
+  } catch (...) {
+    std::cout << "In Test_Blas2_syr, '" << situation
+              << "', ger() call: caught unknown exception" << std::endl;
+    gotUnknownException = true;
+  }
+
+  EXPECT_EQ(gotUnknownException, false)
+      << "Failed test, '" << situation
+      << "': unknown exception should not have happened for ger() call";
+
+  EXPECT_EQ(gotStdException, false)
+      << "Failed test, '" << situation
+      << "': kk ger() should not have thrown a std::exception";
+
+  // ********************************************************************
+  // Prepare h_ger_reference to be compared against h_A_syr
+  // ********************************************************************
+  view_stride_adapter<_ViewTypeExpected, true> h_ger_reference(
+      "h_ger_reference", _M, _N);
+  Kokkos::deep_copy(h_ger_reference.d_base, A_ger.d_base);
+
+  std::string uplo = _useUpOption ? "U" : "L";
+  for (int i = 0; i < _M; ++i) {
+    for (int j = 0; j < _N; ++j) {
+      if (((_useUpOption == true) && (i <= j)) ||
+          ((_useUpOption == false) && (i >= j))) {
+        // Keep h_ger_reference as already computed
+      } else {
+        h_ger_reference.d_view(i, j) = org_A.h_view(i, j);
+      }
+    }
+  }
+  if (_useHermitianOption && _A_is_complex) {
+    for (int i(0); i < _N; ++i) {
+      h_ger_reference.d_view(i, i) =
+          0.5 * (h_ger_reference.d_view(i, i) +
+                 _KAT_A::conj(h_ger_reference.d_view(i, i)));
+    }
+  }
+
+  // ********************************************************************
+  // Compare
+  // ********************************************************************
+  this->compareKkSyrAgainstReference(alpha, h_A_syr, h_ger_reference.d_view);
+}
+
+}  // namespace Test
+
+template <class ScalarX, class ScalarA, class Device>
+int test_syr(const std::string& caseName) {
+  KOKKOS_IMPL_DO_NOT_USE_PRINTF(
+      "+======================================================================="
+      "===\n");
+  KOKKOS_IMPL_DO_NOT_USE_PRINTF("Starting %s ...\n", caseName.c_str());
```

Comment:
I think you are right, Carl. I will leave like this in the test code, so that I have a place to consult once I want to use the PRINTF for debugging other efforts in the future.